static/js/lib/search.js

```diff
@@ -413,37 +414,33 @@ export const buildSearchQuery = ({
 
   const types = getTypes(type)
   for (const type of types) {
-    // One of the text fields must match
-    const matchQuery = text
-      ? {
-        should: [
-          {
-            multi_match: {
-              query:     text,
-              fields:    searchFields(type),
-              fuzziness: "AUTO"
+    let textQuery = {}
+    const textClauses = []
+    let textFilter = []
+    if (!emptyOrNil(text)) {
+      textClauses.push({
+        multi_match: {
+          query:     text,
+          fields:    searchFields(type),
+          fuzziness: "AUTO"
+        }
+      })
+      if ([LR_TYPE_BOOTCAMP, LR_TYPE_COURSE, LR_TYPE_PROGRAM].includes(type)) {
+        textClauses.push({
+          nested: {
+            path:  "runs",
+            query: {
+              multi_match: {
+                query:     text,
+                fields:    RESOURCE_QUERY_NESTED_FIELDS,
+                fuzziness: "AUTO"
+              }
             }
           }
-        ]
+        })
       }
-      : {}
-
-    if (
-      text &&
-      [LR_TYPE_BOOTCAMP, LR_TYPE_COURSE, LR_TYPE_PROGRAM].includes(type)
-    ) {
-      matchQuery.should.push({
-        nested: {
-          path:  "runs",
-          query: {
-            multi_match: {
-              query:     text,
-              fields:    RESOURCE_QUERY_NESTED_FIELDS,
-              fuzziness: "AUTO"
-            }
-          }
-        }
-      })
+      textQuery = { should: textClauses }
+      textFilter = [{ bool: textQuery }]
```

Comment:
just a little thought I had about how this could maybe be reorganized a little bit for clarity. it looks like we have basically that `textFilter === emptyOrNil(text) ? [] : [{ bool: textQuery }]`, if it was declared with `const` after this block you could declare it that way (I'm just sometimes way of the pattern of declaring a variable with `let` and then assigning it many lines away from it's declaration, I think it's best to avoid `let` wherever we can).

it seems also like the `textClauses` variable doesn't need to be declared outside of the `if () { ... }` block. You could even just do something like:

```js
const textQuery = emptyOrNil(text)
  ? {}
  : {
    should: [
      {
        multi_match: {
          query:     text,
          fields:    searchFields(type),
          fuzziness: "AUTO"
        }
      },
      [LR_TYPE_BOOTCAMP, LR_TYPE_COURSE, LR_TYPE_PROGRAM].includes(type)
        ? {
          nested: {
            path:  "runs",
            query: {
              multi_match: {
                query:     text,
                fields:    RESOURCE_QUERY_NESTED_FIELDS,
                fuzziness: "AUTO"
              }
            }
          }
        }
        : null
    ].filter(clause => clause !== null)
  }

const textFilter = emptyOrNil(text) ? [] : [{ bool: textQuery }]
```

to just declare the variables in one go. Anyway, take this or leave it.

Reply:
👍 change made